src/CMakeLists.txt

```diff
@@ -303,91 +399,48 @@ set(src_nupiccore_srcs
     nupic/utils/Watcher.cpp)
 
 set(src_lib_static_nupiccore_srcs
-    ${src_py_support_files}
-    ${src_nupiccore_srcs})
+    ${src_nupiccore_srcs}
+    ${src_py_support_files})
 
-set(src_lib_static_nupiccore_compile_flags
-    "${src_compile_flags} -I${src_numpy_core}/include")
 
-message(STATUS "src_compile_flags = ${src_compile_flags}")
-message(STATUS "src_lib_static_nupiccore_compile_flags = ${src_lib_static_nupiccore_compile_flags}")
+message(STATUS "FOR SOLO: src_compile_flags = ${src_compile_flags}")
 
 add_library(${src_lib_static_nupiccore_solo} STATIC
             ${src_lib_static_nupiccore_srcs})
-# nupic_core_solo sources depend on headers installed by these external projects
-add_dependencies(${src_lib_static_nupiccore_solo}
-                 ${YAML_CPP_STATIC_LIB_TARGET}
-                 ${YAML_STATIC_LIB_TARGET}
-                 ${APR1_STATIC_LIB_TARGET}
-                 ${APRUTIL1_STATIC_LIB_TARGET}
-		 )
 set_target_properties(${src_lib_static_nupiccore_solo} PROPERTIES COMPILE_FLAGS
-                      ${src_lib_static_nupiccore_compile_flags})
+                      ${src_compile_flags})
 if(${NUPIC_IWYU})
-  # TODO: Create a target that doesn't include the generated capnp schema files
-  # since we don't want to run iwyu on them and iwyu can only be applied to
-  # CMake targets (not source files directly).
   set_target_properties(${src_lib_static_nupiccore_solo}
     PROPERTIES CXX_INCLUDE_WHAT_YOU_USE ${iwyu_path})
 endif()
 
 
-#
+##########
 # Create the nupic_core "combined" static library by merging nupic_core_solo
 # with our external static libraries
 #
+set(static_libs
+	${src_external_static_libs}
+	${src_lib_static_nupiccore_solo}
+	)
+include(${REPOSITORY_DIR}/src/NupicLibraryUtils.cmake) # for MERGE_STATIC_LIBRARIES
+set(src_lib_static_nupiccore nupic_core)
+merge_static_libraries(${src_lib_static_nupiccore} "${static_libs}" )
 
-include(src/NupicLibraryUtils) # for MERGE_STATIC_LIBRARIES
-
-set(src_lib_static_nupiccore_combined nupic_core)
-
-set(src_external_static_libs
-    
-    ${YAML_CPP_STATIC_LIB_TARGET}
-    ${YAML_STATIC_LIB_TARGET}
-    ${APR1_STATIC_LIB_TARGET}
-    ${APRUTIL1_STATIC_LIB_TARGET}
-   )
-
-set(src_combined_nupiccore_source_archives
-    ${src_lib_static_nupiccore_solo}
-    ${src_external_static_libs})
-
-# Create a top-level library target for the combined static lib
-merge_static_libraries(${src_lib_static_nupiccore_combined}
-                       "${src_combined_nupiccore_source_archives}")
 
-
-#
+###########
 # Build tests of the nupic_core "combined" static library
 #
 
 # Common libs for test executables
 set(src_common_test_exe_libs
-    ${src_lib_static_nupiccore_combined}
+    ${src_lib_static_nupiccore}
     ${PYTHON_LIBRARIES}
     ${src_common_os_libs})
 
 message(STATUS "src_common_test_exe_libs = ${src_common_test_exe_libs}")
 
 
-## Example custom executable 
```

Comment:
let's keep one executable as an example

Reply:
Hmmm, probably should, but they all went away.